go/tools/extract_package/extract_test.go

```diff
@@ -0,0 +1,21 @@
+package main
+
+import (
+	"testing"
+)
+
+func TestExtract(t *testing.T) {
+	for _, fname := range []string{
+		"extract.go",
+		"extract_test.go",
+		"ignored.go",
```

Comment:
I tried to say that the function can extract a package name even if the file contains build tags before the `package` clause.  It didn't need to be `+build ignored` but it just came to my mind first.